docs/internal/interop.md

```diff
@@ -0,0 +1,351 @@
+# Verona Interoperability / Sandbox
+
+This is an initial attemp at mapping all the requirements that the
+interoperability layer will have in Verona to use external code written in other
+languages (like C++, Rust, etc) in a safe manner (sanbdox).
+
+## Usage in Verona
+
+The aim is to have external code in Verona with a native interface that looks
+and feels like Verona code. We want to abstract away the entirety of the sandbox
+functionality, the foreign function dispatch and serialisations, the execution
+models, etc.
+
+The syntax is still largely undefined, but this is an example of using the C++
+`std::vector` class inside Verona as pseudo-Verona code:
+
+```c
+// include/import the c++ code as a Verona module
+type StdVec = "c++::vector";
+
+// Create a new region (we know it's foreign, so it's a sandbox)
+reg = new StdVec;
```

Comment:
We still need to specify the kind of sandbox, because we're likely to have multiple sandboxing mechanisms available.

Reply:
Not yet, but it will be the same syntax that we use for creating objects in regions of specific kinds.  I think @sylvanc had some straw-man syntax, once the compiler can handle the difference between traced and bump-allocated regions then we will use the same syntax.  The key point is that, from the perspective of this code, creating a new process-based sandboxed region is the same syntactic construct as creating a new region of any other explicit kind.